### PR TITLE
Search updates and homepage redesign

### PR DESCRIPTION
**Pull request by Trombach**
In this pull request:
* New global search bar with filters mega menu
* Redesigned search results page with continuous loading
* Redesigned homepage
* Categories page
* Activities page
* Search Results highlighting
* New layout for mobile navbar

### CODE REVIEW

**Comment by Trombach** [comment]
The new mobile navbar design is now included in this pull request. I've used it to solve some of the button text misalignment issues we've seen in the navbar with the new search bar. I hope that's ok?

File: research-hub-web/src/app/components/layout/search-bar/search-bar.component.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    this.router.navigate(
+      ['/search'],
+      {
+        queryParams: this.searchService.generateQueryParams(this.searchText, this.activeFilters)
+      }
+    );
+  }
+
+  public focus(): void {
+    this.renderer.selectRootElement(this.searchBox.nativeElement).focus();
```
Comment (rosemcc) [inline research-hub-web/src/app/components/layout/search-bar/search-bar.component.ts:83]: Not focusing on mobile view

**Review by rosemcc — APPROVED** [review]

File: research-hub-web/src/styles.scss [hunk 2]
```diff
@@ -199,15 +183,29 @@ mat-expansion-panel {
...
-.search-results-text {
-  color: $dark-blue !important;
-  font-weight: 500;
-  font-style: italic;
-}
 .chip-avatar {
   margin-right: 16px;
   border-radius: 50%;
 }
+.cdk-virtual-scroll-content-wrapper {
```
Comment (Trombach) [inline research-hub-web/src/styles.scss:190]: I think this was only for virtual scrolling on the search results page? Can we remove this?

**Review by Trombach — COMMENTED** [review]
@rosemcc Looks good! I've added a few comments, have a look whether we want to act upon any of them. They are all very minor!

File: research-hub-web/src/app/components/activities-page/activities-page.component.html [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            <div>
+              <a routerLink="/search"
+              [queryParams]="
+                searchService.generateQueryParams('', {
+                  category: [],
+                  stage: [item.sys.id],
+                  relatedOrgs: []
+                })
+              ">
+              <h3>{{ item.name }}</h3>
```
[the remark this answers is not in the document]
Comment (rosemcc) [reply research-hub-web/src/app/components/activities-page/activities-page.component.html:30]: The page title is h2 and there's no h1 element. Yes, let's leave for now and review as part of the accessibility changes.

File: research-hub-web/src/app/components/activities-page/activities-page.component.spec.ts [hunk 4]
```diff
@@ -45,18 +44,19 @@ describe('ResearchActivityComponent', () => {
...
     })
     .compileComponents();
   });
 
   beforeEach(() => {
-    fixture = TestBed.createComponent(ResearchActivityComponent);
+    fixture = TestBed.createComponent(ActivitiesPageComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
```
[the remark this answers is not in the document]
Comment (rosemcc) [reply research-hub-web/src/app/components/activities-page/activities-page.component.spec.ts:62]: I reckon we can delay adding any more unit tests for this, seeing the activities page is going to change later on

File: research-hub-web/src/app/components/activities-page/activities-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  public allStages$: Observable<StageCollection>;
+  private subscriptions: Subscription = new Subscription();
+
+  constructor(
+    public allStagesGQL: AllStagesGQL,
+    private getHomepageGQL: GetHomepageGQL,
+    public searchService: SearchService
+  ) { }
+
+  async ngOnInit() {
```
[the remark this answers is not in the document]
Comment (rosemcc) [reply research-hub-web/src/app/components/activities-page/activities-page.component.ts]: Done :)

File: research-hub-web/src/app/components/categories-page/categories-page.component.html [hunk 6]
```diff
@@ -0,0 +1,27 @@
+<div id="research-categories" *ngIf="allCategories$ | async; let allCategories">
+  <div>
+    <h2>{{ title }}</h2>
```
[the remark this answers is not in the document]
Comment (rosemcc) [reply research-hub-web/src/app/components/categories-page/categories-page.component.html]: @Trombach is it just the extra div? I've removed that now but let me know if there was anything else?

File: research-hub-web/src/app/components/categories-page/categories-page.component.spec.ts [hunk 7]
```diff
@@ -100,18 +100,20 @@ describe('BrowseComponent', () => {
...
     })
     .compileComponents();
   });
 
   beforeEach(() => {
-    fixture = TestBed.createComponent(BrowseComponent);
+    fixture = TestBed.createComponent(CategoriesPageComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
```
[the remark this answers is not in the document]
Comment (rosemcc) [reply research-hub-web/src/app/components/categories-page/categories-page.component.spec.ts:119]: Ditto with the categories page - leave for now seeing it's going to change soon

File: research-hub-web/src/app/components/categories-page/categories-page.component.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  private subscriptions: Subscription = new Subscription();
+
+  constructor(
+    public allCategoriesGQL: AllCategoriesGQL,
+    private getHomepageGQL: GetHomepageGQL,
+    public searchService: SearchService,
+    private router: Router
+  ) { }
+
+  async ngOnInit() {
```
[the remark this answers is not in the document]
Comment (rosemcc) [reply research-hub-web/src/app/components/categories-page/categories-page.component.ts:30]: Done

File: research-hub-web/src/styles.scss (hunk 2 above)
Comment (rosemcc) [reply research-hub-web/src/styles.scss:190]: Removed

File: research-hub-web/src/app/components/layout/search-bar/search-bar.component.ts (hunk 1 above)
Comment (Trombach) [reply research-hub-web/src/app/components/layout/search-bar/search-bar.component.ts:83]: The problem: the search button on mobile, which opens the search interface, retains focus und does not want to let lose of it. Tried to use FocusMonitor to get rid of the focus, but could not get it to work. Merging as is, but we could look at this again in the future. Focus states seem to be problematic across browsers as well...

**Pull request merged**